joblib/test/test_logger.py

```diff
@@ -21,48 +18,26 @@
 except NameError:
     unicode = lambda s: s
 
-###############################################################################
-# Test fixtures
-env = dict()
 
-
-def setup():
-    """ Test setup.
-    """
-    cachedir = mkdtemp()
-    if os.path.exists(cachedir):
-        shutil.rmtree(cachedir)
-    env['dir'] = cachedir
-
-
-def teardown():
-    """ Test teardown.
-    """
-    #return True
-    shutil.rmtree(env['dir'])
-
-
-###############################################################################
-# Tests
-def test_print_time():
+def test_print_time(tmpdir):
     # A simple smoke test for PrintTime.
+    logfile = tmpdir.join('test.log').strpath
     try:
         orig_stderr = sys.stderr
         sys.stderr = io.StringIO()
-        print_time = PrintTime(logfile=os.path.join(env['dir'], 'test.log'))
+        print_time = PrintTime(logfile=logfile)
         print_time(unicode('Foo'))
         # Create a second time, to smoke test log rotation.
-        print_time = PrintTime(logfile=os.path.join(env['dir'], 'test.log'))
+        print_time = PrintTime(logfile=logfile)
         print_time(unicode('Foo'))
         # And a third time
-        print_time = PrintTime(logfile=os.path.join(env['dir'], 'test.log'))
+        print_time = PrintTime(logfile=logfile)
         print_time(unicode('Foo'))
         printed_text = sys.stderr.getvalue()
         # Use regexps to be robust to time variations
         match = r"Foo: 0\..s, 0\..min\nFoo: 0\..s, 0..min\nFoo: " + \
                 r".\..s, 0..min\n"
-        if not re.match(match, printed_text):
-            raise AssertionError('Excepted %s, got %s' %
-                                    (match, printed_text))
+        assert (re.match(match, printed_text),
+                'Excepted %s, got %s' % (match, printed_text))
```

Comment:
@lesteve Ths change was for uniformity as I am revamping the whole test, but please do let me know if the old block of code was used for specific purposes. This one works just fine though.